### PR TITLE
Prep for v1.23.0

### PR DESCRIPTION
**Pull request by odow**
## Pre-release

 - [x] Check that the pinned packages in `docs/Project.toml` are updated. We pin
       the versions so that changes in the solvers (changes in printing, small
       numeric changes) do not break the printing of the JuMP docs in arbitrary
       commits.
 - [x] Check that the `rev` fields in `docs/packages.toml` are updated. We pin
       the versions of solvers and extensions to ensure that changes to their
       READMEs do not break the JuMP docs in arbitrary commits, and to ensure
       that the versions are compatible with the latest JuMP and
       MathOptInterface releases.
 - [x] Check compat of `DimensionalData` in `Project.toml`
 - [x] Check compat of `MacroTools` in `Project.toml`
 - [x] Update `docs/src/changelog.md`
 - [x] <s>https://github.com/jump-dev/JuMP.jl/actions/runs/10261874110</s>
 - [x] https://github.com/jump-dev/JuMP.jl/actions/runs/10360429801
    Lots of breakage
     - [x] Alpine
     - [x] BilevelJuMP
     - [x] InfiniteOpt https://github.com/infiniteopt/InfiniteOpt.jl/pull/361 
     - [x] LinearFractional: issue is on their end https://github.com/nexteraanalytics/LinearFractional.jl/pull/21#discussion_r1705026288. Will be fixed by https://github.com/nexteraanalytics/LinearFractional.jl/pull/22
     - [x] PolyJuMP: @blegat?
     - [x] SDDP: should be fixed by https://github.com/odow/SDDP.jl/pull/770
     - [x] SumOfSquares: @blegat?
 - [x] Change the version number in `Project.toml`
 - [x] The commit messages in this PR do not contain `[ci skip]`

## The release

 - [ ] After merging this pull request, comment `[at]JuliaRegistrator register` in
       the GitHub commit. This should automatically publish a new version to the
       Julia registry, as well as create a tag, and rebuild the documentation
       for this tag.

       These steps can take quite a bit of time (1 hour or more), so don't be
       surprised if the new documentation takes a while to appear. In addition,
       the links in the README will be broken until JuliaHub fetches the new
       version on their servers.

## Post-release

 - [ ] Once the tag is created, update the relevant `release-` branch. The latest
       release branch at the time of writing is `release-1.0` (we haven't
       back-ported any patches that needed to create a `release-1.Y` branch). To
       to update the release branch with the v1.10.0 tag, do:
       ```
       git checkout release-1.0
       git pull
       git merge v1.10.0
       git push
       ```

### CODE REVIEW

File: docs/Project.toml [hunk 1]
```diff
@@ -50,17 +50,17 @@ Dualization = "0.5"
...
-HiGHS = "=1.9.0"
+HiGHS = "=1.9.2"
 Interpolations = "0.14"
-Ipopt = "=1.6.2"
+Ipopt = "=1.6.5"
 JSON = "0.21"
 JSONSchema = "1"
 Literate = "2.8"
-MathOptInterface = "=1.30.0"
+MathOptInterface = "=1.31.0"
```
Comment (odow) [inline docs/Project.toml:59]:
```suggestion
MathOptInterface = "=1.30.0"
```

**Comment by blegat** [comment]
PolyJuMP and SumOfSquares failing can be ignore, they need to checkout a few packages for the tests to work

**Comment by odow** [comment]
@joaquimg BilevelJuMP is:
image

File: docs/Project.toml [hunk 2]
```diff
@@ -50,17 +50,17 @@ Dualization = "0.5"
...
-HiGHS = "=1.9.0"
+HiGHS = "=1.9.2"
 Interpolations = "0.14"
-Ipopt = "=1.6.2"
+Ipopt = "=1.6.5"
 JSON = "0.21"
 JSONSchema = "1"
 Literate = "2.8"
-MathOptInterface = "=1.31.0"
+MathOptInterface = "=1.30.0"
```
Comment (odow) [inline docs/Project.toml:59]:
```suggestion
MathOptInterface = "=1.31.0"
```

File: docs/src/changelog.md [hunk 3]
```diff
@@ -7,6 +7,30 @@ CurrentModule = JuMP
...
+### Other
+
+ - Documentation improvements
+   - Fixed missing character in installation instructions (#3777)
+   - Added a section of querying the Jacobian (#3779)
+   - Clarify that SCIP does not support lazy constraints (#3784)
+   - Fixed typo in `knapsack.jl` (#3792)
+   - Added a warning to docs about tolerances in Bin and Int variables (#3794)
+   - Clarify where to type installation commands (#3795)
+ - Improve error message for common incorrect syntax in constraint macro (#3781)
```
Comment (odow) [inline docs/src/changelog.md:32]:
```suggestion
 - Improve error message for common incorrect syntax in constraint macro (#3781)
 - Changed `show(::IO, ::GenericModel)` to a more informative tree structure (#3803)
```

**Review by blegat — APPROVED** [review]

**Pull request merged**